docs/reference/module-options.md

```diff
@@ -372,3 +372,442 @@ module.exports = {
+<!-- TODO: Change index page link to a guide on index pages when available. -->
+
+#### Example
+
+```javascript
+// modules/article-page/index.js
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    perPage: 12
+  },
+  // ...
+}
+```
+
+### `next`
+
+If set to `true`, Apostrophe will include the next piece, based on the [sort option](#sort), on `req.data.next` when serving a [show page](/reference/glossary.md#show-page). This is useful to add links to a show page directing visitors to the next item in a series (e.g., the next oldest blog post). If not set, `req.data.next` will not be available.
+
+`next` can also be set to an object, which will be used as a query builder for retrieving that next piece document.
+
+#### Example
+
+```javascript
+// modules/article-page/index.js
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    next: true
+  },
+  // ...
+}
+
+// OR
+
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    // The next article piece would be returned with only the `title`, `_url`,
+    // and `publishedAt` properties.
+    next: {
+      project: {
+        title: 1,
+        _url: 1,
+        publishedAt: 1
+      }
+    }
+  },
+  // ...
+}
+```
+
+### `piecesFilters`
+
+<!-- TODO: Link to a guide on using piece filters when available. -->
+<!-- TODO: Link to a better query builder guide when available. -->
+`piecesFilters` can be configured as an array of objects to support filtering pieces on an [index page](/reference/glossary.md#index-page). Each object must have a `name` property associated with a valid [query builder](/guide/major-changes.md#queries). These include:
+
+- Field names whose field types automatically get builders: `string`, `slug`, `boolean`, `checkboxes`, `select`, `integer`, `float`, `url`, `date`, `relationship`
+- Custom query builders configured in an app that include a `launder` method
+
+When the index page is served, configured filters will be represented on a `req.data.piecesFilters` object (`data.piecesFilters` in the template). If you include `counts: true` in a filter object, the number of pieces matching that filter are included on `req.data.piecesFilters` properties.
+
+#### Example
+
+```javascript
+// modules/article-page/index.js
+// 👆
+module.exports = {
+  extend: '@apostrophecms/piece-page-type',
+  options: {
+    piecesFilters: [
+      { name: '_author' },
+      {
+        name: 'category',
+        counts: true
+      }
+    ]
+  },
+  fields: {
+    _author: {
+      type: 'relationship',
+      label: 'Author'
+    },
+    category: {
+      type: 'select',
+      label: 'Category',
+      choices: [
+        // Category choices here
+      ]
+    }
+    // Other fields...
+  },
+  // ...
+}
+```
+
+### `piecesModuleName`
```

Comment:
This option should get renamed `pieceModuleName`.

Reply:
For consistency with our singular-everywhere policy on module names.

Reply:
➕